pkg/messaging/v1/subscriptions_client.go

```diff
@@ -94,6 +102,35 @@ func (c *subscriptionsClient) UpdateSubscription(ctx context.Context, subscripti
 	return knerrors.GetError(err)
 }
 
+func (c *subscriptionsClient) UpdateSubscriptionWithRetry(ctx context.Context, name string, updateFunc SubscriptionUpdateFunc, nrRetries int) error {
+	return updateSubscriptionWithRetry(ctx, c, name, updateFunc, nrRetries)
+}
+
+func updateSubscriptionWithRetry(ctx context.Context, c KnSubscriptionsClient, name string, updateFunc SubscriptionUpdateFunc, nrRetries int) error {
+	b := retry.DefaultRetry
+	b.Steps = nrRetries
+	err := retry.RetryOnConflict(b, func() error {
+		return updateSubscription(ctx, c, name, updateFunc)
+	})
+	return err
+}
+
+func updateSubscription(ctx context.Context, c KnSubscriptionsClient, name string, updateFunc SubscriptionUpdateFunc) error {
+	sub, err := c.GetSubscription(ctx, name)
+	if err != nil {
+		return err
+	}
+	if sub.GetDeletionTimestamp() != nil {
+		return fmt.Errorf("can't update subscription %s because it has been marked for deletion", name)
```

Comment:
I wonder if in this case the retries will still apply ? I mean, this is a terminal error which won't be fixed when trying more often. But I guess that the retry support only retries in case of conflicts, right ?

Reply:
Yes, it will only retry on conflict

Reply:
thanks!